src/QtAVPlayer/qavcodec.cpp

```diff
@@ -67,7 +67,7 @@ bool QAVCodec::open(AVStream *stream)
 
     av_opt_set_int(d->avctx, "refcounted_frames", true, 0);
     av_opt_set_int(d->avctx, "threads", 1, 0);
-    ret = avcodec_open2(d->avctx, d->codec, nullptr);
+    ret = avcodec_open2(d->avctx, d->codec, opts);
```

Comment:
did not check myself, but is it possible to set codec opts after open() ?

Reply:
Do you mean not to put this in open, but to try to set these parameters in a separate function? Or did I misunderstand the question?

Reply:
yes, but looks it is ok to have it during open()